protocol/src/main/java/org/mifos/chatbot/protocol/telegram/TelegramChatService.java

```diff
@@ -0,0 +1,183 @@
+package org.mifos.chatbot.protocol.telegram;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.mifos.chatbot.client.ApiClient;
+import org.mifos.chatbot.core.AdapterService;
+import org.mifos.chatbot.core.model.MifosResponse;
+import org.mifos.chatbot.core.model.MifosSettings;
+import org.mifos.chatbot.database.dao.UserRepository;
+import org.mifos.chatbot.database.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.telegram.telegrambots.bots.TelegramLongPollingBot;
+import org.telegram.telegrambots.meta.TelegramBotsApi;
+import org.telegram.telegrambots.meta.api.methods.send.SendMessage;
+import org.telegram.telegrambots.meta.api.objects.Update;
+import org.telegram.telegrambots.meta.exceptions.TelegramApiException;
+
+import javax.annotation.PostConstruct;
+import javax.sql.DataSource;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.List;
+
+@Slf4j
+@Component
+public class TelegramChatService extends TelegramLongPollingBot {
+
+    @Autowired
+    private MifosSettings settings;
+
+    @Autowired
+    private AdapterService adapterService;
+
+    @Autowired
+    DataSource dataSource;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Value("${telegram.botUsername}")
+    private String botUsername;
+
+    @Value("${telegram.botToken}")
+    private String botToken;
+
+    @Override
+    public void onUpdateReceived(Update update) {
+        if (update.hasMessage() && update.getMessage().hasText()) {
```

Comment:
Modularize this please

Reply:
Same as facebookMessengerService and SlackChatService.

Reply:
Resolved